test/fast/Distributions.test.ts

```diff
@@ -1,22 +1,176 @@
-    it("also stores the creation block");
-    it("delegates a call to balanceOf to the ERC20 token with the correct parameters");
-    it("makes sure that the available balance is exactly equal to the total");
-    it("advances to the FeeSetup phase");
+    describe("with the correct params passed", async () => {
+      let distribution: MockContract<Distribution>;
+
+      beforeEach(async () => {
+        // Stub out the ERC20 contract.
+        erc20.balanceOf.reset();
+        erc20.balanceOf.returns(10);
+
+        // Deploy the Distribution contract.
+        const factory = await smock.mock<Distribution__factory>("Distribution", deployer);
+
+        distribution = await factory.deploy({
+          distributor: governor.address,
+          issuer: issuerMember.address,
+          fast: fast.address,
+          token: erc20.address,
+          total: BigNumber.from(10),
+          blockLatch: BigNumber.from(1)
+        });
+      });
+
+      it("stores its initial parameters", async () => {
+        const originalParams = await distribution.params();
+        const params = abiStructToObj(originalParams) as Distribution.ParamsStruct;
+
+        expect(params).to.eql({
+          distributor: governor.address,
+          issuer: issuerMember.address,
+          fast: fast.address,
+          token: erc20.address,
+          total: BigNumber.from(10),
+          blockLatch: BigNumber.from(1)
+        });
+      });
+
+      it("initializes the `available` funds using the initial `total`", async () => {
+        const subject = await distribution.available();
+        expect(subject).to.eq(BigNumber.from(10));
+      });
+
+      it("stores the creation block", async () => {
+        const latestBlockNumber = (await ethers.provider.getBlock("latest")).number;
+        const subject = await distribution.creationBlock();
+        expect(subject).to.eq(BigNumber.from(latestBlockNumber));
+      });
+
+      it("delegates a call to balanceOf to the ERC20 token with the correct parameters", async () => {
+        // TODO: Not super accurate... we've had multiple calls at this point.
+        expect(erc20.balanceOf).to.have.been.calledWith(distribution.address);
+      });
+
+      it("advances to the FeeSetup phase", async () => {
+        const subject = await distribution.phase();
+        expect(subject).to.eq(DistributionPhase.FeeSetup);
+      });
+    });
+
+    describe("reversions", async () => {
+      it("reverts if the latched block is in the future", async () => {
+        const totalFunds = 10;
+        const latestBlockNumber = (await ethers.provider.getBlock("latest")).number;
+
+        // Stub out the ERC20 contract.
+        erc20.balanceOf.reset();
+        erc20.balanceOf.returns(totalFunds);
+
+        // Deploy the Distribution contract.
+        const factory = await smock.mock<Distribution__factory>("Distribution", deployer);
+
+        const distribution = factory.deploy({
```

Comment:
Since you are using `factory.deploy` many times, you could have a `const deployDistribution = (params) => {...}` within your parent `describe` block, so that each test could simply `deployDistribution(...)`?